creator-node/src/components/tracks/trackContentUploadManager.js

```diff
@@ -0,0 +1,326 @@
+   * @param {Object} logContext
+   * @param {Object} transcodeAndSegmentParams
+   * @param {string} transcodeAndSegmentParams.fileName the file name of the uploaded track (<cid>.<file type extension>)
+   * @param {string} transcodeAndSegmentParams.fileDir the dir path of the temp track artifacts
+   * @returns an Object with the structure: 
+   * {
+      transcodeFilePath {string}: the path to the transcode file,
+      segmentFileNames {string[]}: a list of segment file names; will have the structure `segment<number>.ts`,
+      segmentFilePaths {string[]}: a list of segment file paths,
+      m3u8FilePath {string}: the path to the m3u8 file,
+      fileName {string}: the original upload track file name
+    } 
+   */
+  static async transcodeAndSegment({ logContext }, { fileName, fileDir }) {
+    let transcodeFilePath, segmentFileNames, segmentFilePaths, m3u8FilePath
+    try {
+      const response = await Promise.all([
+        TranscodingQueue.segment(fileDir, fileName, { logContext }),
+        TranscodingQueue.transcode320(fileDir, fileName, { logContext })
+      ])
+
+      segmentFileNames = response[0].segments.fileNames
+      segmentFilePaths = response[0].segments.filePaths
+      m3u8FilePath = response[0].m3u8FilePath
+      transcodeFilePath = response[1].transcodeFilePath
+    } catch (err) {
+      // Prune upload artifacts
+      FileManager.removeTrackFolder({ logContext }, fileDir)
+
+      throw new Error(err.toString())
+    }
+
+    return {
+      transcodeFilePath,
+      segmentFileNames,
+      segmentFilePaths,
+      m3u8FilePath,
+      fileName
+    }
+  }
+
+  /**
+   * 1. Batch saves the transcode and segments to IPFS and to local dir
+   * 2. Fetches segment duration to build a map of <segment CID: duration>
+   * 3. Adds transcode and segments to DB in Files table
+   * 4. Removes the upload artifacts after successful processing
+   * @param {Object} logContext
+   * @param {Object} param
+   * @param {Object} param.session an object that will contain {cnodeUserUUID {string}: the current user's cnodeUserUUID}
+   * @param {string} param.fileName the filename of the uploaded artifact
+   * @param {string} param.fileDir the directory of where the filename is uploaded
+   * @param {string} param.fileDestination the folder to which the uploaded file has been saved	to
+   * @param {string} param.transcodeFilePath the path of where the transcode exists
+   * @param {string[]} param.segmentFileNames a list of segment file names from the segmenting job
+   * @returns {Object}
+   *  returns
+   *  {
+   *    trancodedTrackCID {string} : the transcoded track's CID representation,
+   *    transcodedTrackUUID {string} : the transcoded track's UUID in the Files table,
+   *    track_segments {string[]} : the list of segments CIDs,
+   *    source_file {string} : the filename of the uploaded artifact
+   *  }
+   */
+  static async processTranscodeAndSegments(
+    { logContext },
+    {
+      session,
```

Comment:
why not pass in cnodeUserUUID directly here?

Reply:
touché